frontend/src/components/common/header/header-container.tsx

```diff
@@ -119,21 +120,39 @@ const HeaderContainer: FC = () => {
     handleInputSearch(currentTarget.value);
   };
 
+  const setFocusOnSearchInput = (): void => {
+    searchInputEl.current?.focus();
+  };
+
   const handleClearInputSearch = (): void => {
     handleInputSearch('');
-    searchInputEl.current?.focus();
+    setFocusOnSearchInput();
   };
 
-  const handleSubmitSearch = (e: FormEvent<HTMLFormElement>): void => {
-    e.preventDefault();
+  const handleClickSearchMobileToggle = (): void => {
+    if (!mobileSearchToggle) {
+      setTimeout(setFocusOnSearchInput, 0);
+    }
+    setMobileSearchToggle(!mobileSearchToggle);
+  };
+
+  const handleClickSearchBtn = (): void => {
     if (searchValue) {
       handleSetInputSearch(searchValue);
       handleClearActiveFilterIds();
       const searchUrlParams = new URLSearchParams({ [SearchQueryParam.SEARCH_TEXT]: searchValue });
       navigate(`${AppRoutes.SEARCH}?${searchUrlParams.toString()}`);
+      setMobileSearchToggle(false);
+    } else {
+      searchInputEl.current?.focus();
```

Comment:
why don't you use setFocusOnSearchInput defined above?

Reply:
did not notice)